### PR TITLE
#63 Adjusting stream metadata generator to keep metadata per entry class

### LINKED ISSUE #63: Persist source and sink streams per entry class when processing annotations
Currently, all `@SourceStreams` and `@SinkStreams` annotations present in the jar are processed and the union of the streams is written to the metadata file. If there are multiple entry classes, there is no distinction between their respective source/sink streams. We should retain the information about the entry class for use downstream.

### PR DESCRIPTION
**Pull request by gunnarmorling**
Fixes #63. @nicoweidner, just saw after the fact that there was a draft PR for this one already. Gonna close yours, as this one here adheres to the format expected by the backend by now.

### CODE REVIEW

**Comment by nicoweidner** [comment]
> Fixes #63. @nicoweidner, just saw after the fact that there was a draft PR for this one already. Gonna close yours, as this one here adheres to the format expected by the backend by now.

Yeah, that's fine, the other one was on hold while we figure out the desired format on the monorepo side

File: sdk/src/test/java/co/decodable/sdk/pipeline/internal/config/metadata/MetadataProcessorTest.java [hunk 1]
```diff
@@ -110,10 +145,14 @@ var record = handler.getRecords().get(0);
     assertThat(record.getMessage()).contains("Neither source nor sink streams were declared");
   }
 
-  private static Compilation compile(URL fileWithoutAnnotations) {
+  private static Compilation compile(URL... files) {
+    Arrays.stream(files).map(f -> JavaFileObjects.forResource(f)).collect(Collectors.toSet());
```
Comment (nicoweidner) [inline sdk/src/test/java/co/decodable/sdk/pipeline/internal/config/metadata/MetadataProcessorTest.java:149]: This line seems redundant

**Review by nicoweidner — APPROVED** [review]
LGTM. I think there is one redundant line in a test though

File: sdk/src/test/java/co/decodable/sdk/pipeline/internal/config/metadata/MetadataProcessorTest.java (hunk 1 above)
Comment (gunnarmorling) [reply sdk/src/test/java/co/decodable/sdk/pipeline/internal/config/metadata/MetadataProcessorTest.java:149]: How odd, I guess some left-over from experimentation with the streams API. Good catch!

Comment (gunnarmorling) [inline sdk/src/test/java/co/decodable/sdk/pipeline/internal/config/metadata/MetadataProcessorTest.java:149]:
```suggestion
```

**Pull request merged**